secp256k1.nim

```diff
@@ -607,26 +601,26 @@ func recover*(sig: SkRecoverableSignature, msg: SkMessage): SkResult[SkPublicKey
 
   ok(SkPublicKey(data: data))
 
-func ecdh*(seckey: SkSecretKey, pubkey: SkPublicKey): SkEcdhSecret =
+func ecdh*(seckey: SkSecretKey, pubkey: SkPublicKey): SkResult[SkEcdhSecret] =
   ## Calculate ECDH shared secret.
   var secret {.noinit.}: array[SkEdchSecretSize, byte]
-  let res = secp256k1_ecdh(
+  if secp256k1_ecdh(
```

Comment:
this call can only fail if the provided inputs are invalid but we've already validated the inputs (via `requiresInit` etc) meaning the `doAssert` is appropriate here

Reply:
right, I forget about that, will fix it.

Reply:
notably though, this only applies when using the default hash function - a "failing" hash function would be another failure mode which we need to allow.

The question then becomes: do we return SkResult for the `ecdh` taking a custom hashing function? do we give it another name because it can fail? wdyt?

Reply:
our use case in nim-eth  is using always success hashing function, but future use case may be different.
I think some documentation is enough, no need to give it another name. The SkResult return type itself is enough to raise user awareness that this function may fail. The implementer of the custom hash function can decide further action.